presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveViews.java

```diff
@@ -231,6 +233,45 @@ public void testHiveViewInInformationSchema()
                 .contains(row("n_nationkey", "bigint", "", ""));
     }
 
+    @Test(groups = HIVE_VIEWS)
+    public void testFailingHiveViewsForInformationSchema()
+    {
+        if (getHiveVersionMajor() < 3) {
+            // Skip older versions since they don't support "getTableNamesByType" in ThriftHiveMetastore#getAllViews
+            throw new SkipException("Only test the behavior for hive 3");
+        }
+
+        onHive().executeQuery("DROP SCHEMA IF EXISTS test_list_failing_views CASCADE");
+        onHive().executeQuery("CREATE SCHEMA test_list_failing_views");
+        onHive().executeQuery("CREATE VIEW test_list_failing_views.correct_view AS SELECT * FROM nation limit 5");
+
+        // Create a view for which the translation is guaranteed to fail
+        onPresto().executeQuery("CREATE TABLE test_list_failing_views.table_dropped (col0 BIGINT)");
+        onHive().executeQuery("CREATE VIEW test_list_failing_views.failing_view AS SELECT * FROM test_list_failing_views.table_dropped");
+        onPresto().executeQuery("DROP TABLE test_list_failing_views.table_dropped");
+
+        assertThat(query("SELECT table_name FROM information_schema.views WHERE table_schema = 'test_list_failing_views'"))
+                .containsOnly(row("correct_view"));
+
+        assertThat(query("SELECT table_name FROM information_schema.views"))
+                .contains(row("correct_view"));
+
+        assertThat(query("SELECT table_name FROM information_schema.views WHERE table_schema = 'test_list_failing_views' and table_name = 'correct_view'"))
+                .containsOnly(row("correct_view"));
+
+        // Listing fails when metadata for the problematic view is queried specifically
+        assertThatThrownBy(() -> query("SELECT table_name FROM information_schema.views WHERE table_schema = 'test_list_failing_views' and table_name = 'failing_view'"))
+                .hasMessageContaining("Failed to translate Hive view 'test_list_failing_views.failing_view'");
+
+        // Queries on information_schema.columns also trigger ConnectorMetadata#getViews. Columns from failing_view are
+        // listed too since HiveMetadata#listTableColumns does not ignore views.
+        assertThat(query("SELECT DISTINCT table_name FROM information_schema.columns WHERE table_schema = 'test_list_failing_views'"))
+                .containsOnly(row("correct_view"), row("failing_view"));
+
+        assertThatThrownBy(() -> query("SELECT * FROM information_schema.columns WHERE table_schema = 'test_list_failing_views' AND table_name = 'failing_view'"))
+                .hasMessageContaining("Failed to translate Hive view 'test_list_failing_views.failing_view'");
```

Comment:
Why does it fail? In the unfiltered `information_schema.columns` query we were OK to use information available in metastore, without translating the view. Why are we not doing this here?

Reply:
This is because the unfiltered version uses `HiveMetadata#getViews`, which ignores failures. Filtered version uses `HiveMetadata#getView` which doesn't.

Reply:
That's technically understood and what i anticipated, but is there a sound reason for the difference?

Reply:
Note that `information_schema.views` also has a similar discrepancy of ignoring v/s failing for unfiltered v/s filtered case. Do you think the discrepancy is okay there but not for `information_schema.columns`?

An argument in favor of the discrepancy is that - we want to show most other metadata (from valid views) when only some of the tables/views have invalid metadata. In the "filtered" case, the specific view being looked for is problematic, so failing doesn't result in losing any metadata.

To make filtered v/s unfiltered cases more consistent, I think it'd help to make `ConnectorMetadata#getViews` (and other listing calls) return an iterator, but may be do such an SPI change in a separate PR.

Reply:
From SQL perspective, I see how this is problematic. While listing with a coerce filter, we ignore the failing view. But for point-lookup, we fail the query. So to be consistent, we might want to always ignore so that we can atleast return partial correct metadata where we can. How about we swallow PrestoExceptions from https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java#L1036 to do this? This will resolve discrepancy for both the tables.

Reply:
I think you made a valid point and i see how inconsistent it is. It is not limited to views either and we have this "ignore or throw? sacrifice SQL consistency?" challenge in other places as well.

I think it does not help to solve those cases on one-by-one basis, as this may lead to different decisions in different cases (eg due to different priorities or whatever else). 
Let's maybe have a dedicated discussion, document the outcome somewhere and go ahead. 
My prediction is that chances are high we come to the conclusion aligned with your thinking, but i would really want to preserve the consistency by being a bit more structured about the decision making process.
What do you think?

Reply:
@findepi Does it seem reasonable to you to go ahead with this PR in the current form? We can update this test for filtered case in future according to the final verdict in #6551. (The current approach seems aligned with @kokosing 's opinion too.)